app/src/main/java/org/simple/clinic/teleconsultlog/shareprescription/TeleconsultSharePrescriptionInit.kt

```diff
@@ -0,0 +1,21 @@
+package org.simple.clinic.teleconsultlog.shareprescription
+
+import com.spotify.mobius.First
+import com.spotify.mobius.Init
+
+class TeleconsultSharePrescriptionInit : Init<TeleconsultSharePrescriptionModel, TeleconsultSharePrescriptionEffect> {
+
+  override fun init(model: TeleconsultSharePrescriptionModel): First<TeleconsultSharePrescriptionModel, TeleconsultSharePrescriptionEffect> {
+    val effects = mutableSetOf<TeleconsultSharePrescriptionEffect>()
+
+    if (model.hasPatientProfile.not()) {
+      effects.addAll(listOf(
+          LoadPatientProfile(model.patientUuid),
+          LoadPatientMedicines(model.patientUuid),
+          LoadSignature,
+          LoadMedicalRegistrationId))
+    }
```

Comment:
I just realised that we cannot use the presence of the `PatientProfile` in the model to decide whether to load all the effects since each of them is loaded individually.

i.e, if the model is saved *AFTER* the patient profile is loaded but *BEFORE* any of the other details are loaded and then the screen is restored, the effects to load the medicines, signature, and medical registration ID will never be emitted and the screen will remain partially loaded.

Can we correct this to check the individual pieces of data separately?